### PR TITLE
Upgrading material-ui to @material-ui/core

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
There is a security alert for lodash.merge, which is a dependency for `material-ui`.
`material-ui` was deprecated.

#### What's this PR do?
Upgrades material-ui to `@material-ui/core`

#### How should this be manually tested?
Nothing should break

### CODE REVIEW

**Comment by noisecapella** [comment]
I'm not sure what's going on with Travis. I'll take a look later at the warning and see if I can help figure it out.

**Comment by noisecapella** [comment]
The CSS in the navbar is acting strange. This PR:
Screenshot from 2019-09-10 16-53-58

On production:
Screenshot from 2019-09-10 16-54-09

**Comment by noisecapella** [comment]
The profile edit dialog has many style differences, and there is no save or cancel button:
Screenshot from 2019-09-10 16-59-44

**Comment by noisecapella** [comment]
Actually, it appears that only the edit name and about text dialogs have this problem. The image editing, employer and work history dialogs look a little different but they are fully functional

This PR:
Screenshot from 2019-09-10 17-04-39


On production:
Screenshot from 2019-09-10 17-04-49

**Comment by noisecapella** [comment]
The enroll dialog on the dashboard page is broken:

Screenshot from 2019-09-10 17-29-48

**Comment by noisecapella** [comment]
Those are the issues I found at a glance. Since some of the dialogs appear functional I expect that fixing the other ones should be relatively easy to do.

File: static/js/components/EducationTab.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ class EducationTab extends React.Component {
     dispatch: Function
   }
 
-  componentWillMount() {
+  componentDidMount() {
```
Comment (noisecapella) [inline static/js/components/EducationTab.js]: What was the purpose of this change?

File: static/js/components/PrivacyForm.js [hunk 2]
```diff
@@ -12,17 +13,24 @@ import type {
 } from "../flow/profileTypes"
 import type { UIState } from "../reducers/ui"
 import type { Validator } from "../lib/validation/profile"
+import { sendFormFieldEvent } from "../lib/google_analytics"
+import type { Option } from "../flow/generalTypes"
```
Comment (noisecapella) [inline static/js/components/PrivacyForm.js]: Can you move this and the other `import type` lines to a group below the other imports?

File: static/js/components/SpinnerButton.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ export default class SpinnerButton extends React.Component {
     }
   }
 
-  componentWillReceiveProps(nextProps: SpinnerButtonProps) {
+  getDerivedStateFromProps(nextProps: SpinnerButtonProps) {
```
Comment (noisecapella) [inline static/js/components/SpinnerButton.js]: This is causing a console warning. Can you fix?

Screenshot from 2019-09-10 17-22-11

File: static/js/components/email/EmailCompositionDialog.js [hunk 4]
```diff
@@ -84,7 +84,7 @@ export default class EmailCompositionDialog extends React.Component {
     this.state = editorStateFromProps(props)
   }
 
-  componentWillReceiveProps(nextProps: EmailDialogProps) {
+  getDerivedStateFromProps(nextProps: EmailDialogProps) {
```
Comment (noisecapella) [inline static/js/components/email/EmailCompositionDialog.js]: Same comment as above. Does this need to be a static method?
Screenshot from 2019-09-10 17-22-11

File: static/js/components/inputs/DateField.js [hunk 5]
```diff
@@ -147,57 +147,47 @@ export default class DateField extends React.Component {
     }
 
     const edit = getEditObject()
+    const error = _.get(errors, keySet, undefined) !== undefined
```
Comment (noisecapella) [inline static/js/components/inputs/DateField.js]: Is there no option for errorText anymore?

File: static/js/components/inputs/SelectField.js [hunk 6]
```diff
@@ -91,8 +91,8 @@ class SelectField extends React.Component {
 
   label = (label: string, element: React$Element<*>) => (
     <label className="react-select-label">
-      {label}
       {element}
+      {label}
```
Comment (noisecapella) [inline static/js/components/inputs/SelectField.js]: What was the reason for this change?

File: static/js/components/EducationTab.js (hunk 1 above)
Comment (annagav) [reply static/js/components/EducationTab.js]: componentWillMount is getting deprecated in v16.3.

File: static/js/components/inputs/DateField.js (hunk 5 above)
Comment (annagav) [reply static/js/components/inputs/DateField.js]: I didn't find an equivalent for errorText.

**Comment by annagav** [comment]
@noisecapella I address your comments. Thanks!

File: static/js/components/EducationTab.js (hunk 1 above)
Comment (noisecapella) [reply static/js/components/EducationTab.js]: Ah, good to know

**Comment by noisecapella** [comment]
There's a difference on the 'Leave a MicroMasters program' dialog. Not sure how critical this is for this PR, what do you think @pdpinch @abdulkdawson?

This PR:
Screenshot from 2019-10-17 14-30-31

Master:
Screenshot from 2019-10-17 14-29-02

**Comment by noisecapella** [comment]
There's a difference in the delete dialogs:

This PR:
Screenshot from 2019-10-17 14-39-57


master:
Screenshot from 2019-10-17 14-40-56

**Comment by noisecapella** [comment]
Other than that... everything is looking great. Some minor differences compared to master like font color, to be expected I think.

**Comment by noisecapella** [comment]
I figured out the webpack error. We had a similar problem in xPRO. Can you go to `webpack.config.shared.js` and at the bottom:
 - remove 

        "ignore": [
            "node_modules/**"
          ],

 - remove `exclude: /node_modules/`
 - right after the `test: /\.jsx?$/` line, add:

        include: [
	    path.resolve(__dirname, "static/js"),
	    path.resolve(__dirname, "node_modules/@material-ui"),
        ],

**Comment by noisecapella** [comment]
Sorry to be the bearer of bad news, but it seems like you need to upgrade enzyme to 3.x to work with React 16, and that will be significant extra effort. Functionality looks great though. This is an incredible effort and once we upgrade enzyme and whatever other necessary dependencies, and do the work to update the tests and configuration, I think we could merge this.

**Comment by annagav** [comment]
@pdpinch, I upgraded enzyme, now the tests are executing successfully. I think it says there are 142 failures.

**Comment by noisecapella** [comment]
There are some font styles which look strange:
Screenshot from 2020-01-21 11-03-17

**Comment by noisecapella** [comment]
The styling changes look good to me. Let's merge this!

**Pull request merged**